runtime/queries/metricsview_comparison_toplist.go

```diff
@@ -1045,51 +1126,55 @@ func timeRangeClause(tr *runtimev1.TimeRange, mv *runtimev1.MetricsViewSpec, dia
 	return clause, nil
 }
 
-func validateSort(sorts []*runtimev1.MetricsViewComparisonSort) error {
+func validateSort(sorts []*runtimev1.MetricsViewComparisonSort, comparisonMap map[string]bool, compareQuery bool) error {
 	if len(sorts) == 0 {
 		return fmt.Errorf("sorting is required")
 	}
 	firstSort := sorts[0].Type
 
 	for _, s := range sorts {
 		if firstSort != s.Type {
-			return fmt.Errorf("diffirent sort types are not supported in a single query")
+			return fmt.Errorf("different sort types are not supported in a single query")
 		}
-	}
-	return nil
-}
-
-func isTimeRangeNil(tr *runtimev1.TimeRange) bool {
-	return tr == nil || (tr.Start == nil && tr.End == nil)
-}
-
-func updateComparisonSort(sort []*runtimev1.MetricsViewComparisonSort) {
-	for _, comparisonSort := range sort {
-		if comparisonSort.SortType == runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_UNSPECIFIED && comparisonSort.Type != runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_UNSPECIFIED {
-			switch comparisonSort.Type {
+		// Update sort to make sure it is backwards compatible
+		if s.SortType == runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_UNSPECIFIED && s.Type != runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_UNSPECIFIED {
+			switch s.Type {
 			case runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_BASE_VALUE:
-				comparisonSort.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_BASE_VALUE
+				s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_BASE_VALUE
 			case runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_COMPARISON_VALUE:
-				comparisonSort.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_COMPARISON_VALUE
+				s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_COMPARISON_VALUE
 			case runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_ABS_DELTA:
-				comparisonSort.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_ABS_DELTA
+				s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_ABS_DELTA
 			case runtimev1.MetricsViewComparisonSortType_METRICS_VIEW_COMPARISON_SORT_TYPE_REL_DELTA:
-				comparisonSort.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_REL_DELTA
+				s.SortType = runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_REL_DELTA
+			}
+		}
+
+		if s.SortType == runtimev1.MetricsViewComparisonMeasureType_METRICS_VIEW_COMPARISON_MEASURE_TYPE_UNSPECIFIED {
+			return fmt.Errorf("sort measure '%s' does not have a sort type", s.Name)
+		}
```

Comment:
I think previously you could pass an unspecified sort type?

Reply:
yes I am disallowing that as unlike alias, its a sort measure so makes sense to have a valid sort type, also catches some edge cases as described above. Unless we plan to use it for non measures it makes sense.

Reply:
reverted the change